### PR TITLE
Alberta base personal credit

### LINKED ISSUE #376: Alberta base personal credit
reference: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf

### PR DESCRIPTION
**Pull request by SirMalamute**
Fixes #376

### CODE REVIEW

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    value_type = float
+    entity = Person
+    label = "Alberta base personal credit"
+    unit = CAD
+    definition_period = YEAR
+    reference = "https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf"
+    defined_for = ProvinceCode.AB
+
+    def formula(person, period, parameters):
+        base = parameters(period).gov.provinces.ab.tax.income.credits.base
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py:14]:
```suggestion
        return parameters(period).gov.provinces.ab.tax.income.credits.base
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this maximum amount under the personal credit. 
+values:
+  2023-01-01: 21_003
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta base personal credit maximum amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml:7]:
```suggestion
  label: Alberta basic personal amount
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml:1]: make folder called basic_personal_amount and rename this file to amount.yaml

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this maximum amount under the personal credit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml:1]:
```suggestion
description: Alberta provides this basic personal amount. 
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this maximum amount under the personal credit. 
+values:
+  2023-01-01: 21_003
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml:3]: add 2022 value

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta base personal credit maximum amount
+  reference:
+  - title: Alberta income tax personal credit
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml:9]: add 2022 reference + legal code

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+from policyengine_canada.model_api import *
+
+
+class ab_base_personal_credit(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py:4]:
```suggestion
class ab_basic_personal_amount(Variable):
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: add the credit to the list of non refundable credits

File: changelog_entry.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Alberta base personal income credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Alberta basic personal amount.
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    value_type = float
+    entity = Person
+    label = "Alberta base personal credit"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py:7]:
```suggestion
    label = "Alberta basic personal amount"
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/tests/gov/provinces/ab/tax/income/credits/base.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/ab/tax/income/credits/base.yaml:1]: rename test file to ab_basic_personal_amount.yaml

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py (hunk 6 above)
Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py:4]: also rename var

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/base.yaml:1]: make a `basic_personal_amount` folder and rename this to amount.yaml

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    unit = CAD
+    definition_period = YEAR
+    reference = "https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf"
+    defined_for = ProvinceCode.AB
+
+    def formula(person, period, parameters):
+        return parameters(period).gov.provinces.ab.tax.income.credits.base
+        return base
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_base_personal_credit.py:15]:
```suggestion
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/basic_personal_amount/amount.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+description: Alberta provides this basic personal amount. 
+values:
+  2022-01-01: 19_369
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/basic_personal_amount/amount.yaml:3]:
```suggestion
  # Legal code defines $19,369.
  # Section 44.2 defines indexing.
  2022-01-01: 19_369
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**